src/core/server/csp/config.ts

```diff
@@ -7,28 +7,145 @@
  */
 
 import { TypeOf, schema } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '../internal_types';
+
+interface DirectiveValidationOptions {
+  allowNone: boolean;
+  allowNonce: boolean;
+}
+
+const getDirectiveValidator = (options: DirectiveValidationOptions) => {
+  const validateValue = getDirectiveValueValidator(options);
+  return (values: string[]) => {
+    for (const value of values) {
+      const error = validateValue(value);
+      if (error) {
+        return error;
+      }
+    }
+  };
+};
+
+const getDirectiveValueValidator = ({ allowNone, allowNonce }: DirectiveValidationOptions) => {
+  return (value: string) => {
+    if (!allowNonce && value.startsWith('nonce-')) {
+      return `using "nonce-*" is considered insecure and is not allowed`;
+    }
+    if (!allowNone && (value === `none` || value === `'none'`)) {
+      return `using "none" would conflict with Kibana's default csp configuration and is not allowed`;
+    }
+  };
+};
+
+const configSchema = schema.object(
+  {
+    rules: schema.maybe(schema.arrayOf(schema.string())),
+    script_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    worker_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    style_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
+    }),
+    connect_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+    }),
+    default_src: schema.arrayOf(schema.string(), {
+      defaultValue: [],
+      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
```

Comment:
Wouldn't allowing `'none'` for default-src, connect-src, font-src, and img-src break Kibana?

Functionally it doesn't make a difference since we also add `'self'` to these directives, but as mentioned before we should probably not allow `'none'` and `'self'` to be set at the same time, in case different browsers have different ways of interpreting the CSP spec.
Also, we have #94626 coming down the pike, TL;DR we will need to allow Kibana to frame itself, and we should probably get ahead of that now WRT frame-ancestors and frame-src.

So I think we should make the following changes:

```diff
diff --git a/src/core/server/csp/config.ts b/src/core/server/csp/config.ts
index d8d723ed941..f2075c942b7 100644
--- a/src/core/server/csp/config.ts
+++ b/src/core/server/csp/config.ts
@@ -54,27 +54,27 @@ const configSchema = schema.object(
     }),
     connect_src: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     default_src: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     font_src: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     frame_src: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     img_src: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     frame_ancestors: schema.arrayOf(schema.string(), {
       defaultValue: [],
-      validate: getDirectiveValidator({ allowNone: true, allowNonce: false }),
+      validate: getDirectiveValidator({ allowNone: false, allowNonce: false }),
     }),
     report_uri: schema.arrayOf(schema.string(), {
       defaultValue: [],
diff --git a/src/core/server/csp/csp_directives.ts b/src/core/server/csp/csp_directives.ts
index efb93868a9e..bdb5df1c3b1 100644
--- a/src/core/server/csp/csp_directives.ts
+++ b/src/core/server/csp/csp_directives.ts
@@ -39,6 +39,8 @@ export const additionalRules: Partial<Record<CspDirectiveName, string[]>> = {
   'default-src': [`'self'`],
   'font-src': [`'self'`],
   'img-src': [`'self'`],
+  'frame-ancestors': [`'self'`],
+  'frame-src': [`'self'`],
 };
 
 export class CspDirectives {
```

Reply:
Good catch, did the validation before adding the behavior to add `self` to those directives